### PR TITLE
[csm] strict typing && legacy code removal

### PR DESCRIPTION
**Pull request by jrushlow**
- drops CSM annotation support
- cleans up legacy CSM code
- implements strict typing

- [x] Depends on #1126

### CODE REVIEW

File: src/Util/ClassSourceManipulator.php [hunk 1]
```diff
@@ -97,20 +109,14 @@ public function addEntityField(string $propertyName, array $columnOptions, array
...
-        }
+        $attributes[] = $this->buildAttributeNode(Column::class, $columnOptions, 'ORM');
 
         $defaultValue = null;
         if ('array' === $typeHint) {
             $defaultValue = new Node\Expr\Array_([], ['kind' => Node\Expr\Array_::KIND_SHORT]);
         }
 
-        $this->addProperty($propertyName, $comments, $defaultValue, $attributes);
+        $this->addProperty(name: $propertyName, defaultValue: $defaultValue, attributes: $attributes, comments: $comments);
```
Comment (OskarStark) [inline src/Util/ClassSourceManipulator.php:119]: what about:
```suggestion
        $this->addProperty(
        name: $propertyName,
        defaultValue: $defaultValue,
        attributes: $attributes,
        comments: $comments,
    );
```

File: src/Util/ClassSourceManipulator.php [hunk 2]
```diff
@@ -46,29 +45,23 @@ final class ClassSourceManipulator
...
-    public function __construct(string $sourceCode, bool $overwrite = false, bool $useAnnotations = true, bool $fluentMutators = true, bool $useAttributesForDoctrineMapping = false)
-    {
-        $this->overwrite = $overwrite;
-        $this->useAnnotations = $useAnnotations;
-        $this->fluentMutators = $fluentMutators;
-        $this->useAttributesForDoctrineMapping = $useAttributesForDoctrineMapping;
+    public function __construct(
+        private string $sourceCode,
+        private bool $overwrite = false,
+        private bool $fluentMutators = true,
```
Comment (weaverryan) [inline src/Util/ClassSourceManipulator.php:62]: Do we now EVER use this with `false`?

File: src/Maker/MakeUser.php [hunk 3]
```diff
@@ -147,11 +147,10 @@ public function generate(InputInterface $input, ConsoleStyle $io, Generator $gen
...
         $manipulator = new ClassSourceManipulator(
-            $this->fileManager->getFileContents($classPath),
-            true,
-            !$useAttributesForDoctrineMapping,
-            true,
-            $useAttributesForDoctrineMapping
+            sourceCode: $this->fileManager->getFileContents($classPath),
+            overwrite: true,
+            fluentMutators: true,
+            useAttributesForDoctrineMapping: $useAttributesForDoctrineMapping
```
Comment (weaverryan) [inline src/Maker/MakeUser.php:153]: Is this still valid? Like... when would `$useAttributesForDoctrineMapping` be false... XML? annotations? If annotations, should we throw an exception? If XML... should we throw an exception?

File: src/Util/ClassSourceManipulator.php [hunk 4]
```diff
@@ -97,20 +90,19 @@ public function addEntityField(string $propertyName, array $columnOptions, array
...
             $defaultValue = new Node\Expr\Array_([], ['kind' => Node\Expr\Array_::KIND_SHORT]);
         }
 
-        $this->addProperty($propertyName, $comments, $defaultValue, $attributes);
+        $this->addProperty(
+        name: $propertyName,
+        defaultValue: $defaultValue,
+        attributes: $attributes,
+        comments: $comments,
+    );
```
Comment (weaverryan) [inline src/Util/ClassSourceManipulator.php:105]: indentation got weird here

File: src/Util/ClassSourceManipulator.php [hunk 5]
```diff
@@ -130,29 +122,15 @@ public function addEmbeddedEntity(string $propertyName, string $className): void
...
-            ];
-        } else {
-            $attributes = [
+        $attributes = [
                 $this->buildAttributeNode(
                     Embedded::class,
                     ['class' => new ClassNameValue($className, $typeHint)],
                     'ORM'
                 ),
             ];
```
Comment (weaverryan) [inline src/Util/ClassSourceManipulator.php:131]: I think indentation is weird here too now?

Comment (jrushlow) [reply src/Util/ClassSourceManipulator.php:131]: done

File: src/Maker/MakeUser.php (hunk 3 above)
Comment (jrushlow) [reply src/Maker/MakeUser.php:153]: yes - well kinda, we still need to be able to generate user _models_ if the class isn't an entity. I've added a legacy && todo notation a few lines above to fix this in the pr im working on now.

File: src/Util/ClassSourceManipulator.php (hunk 2 above)
Comment (jrushlow) [reply src/Util/ClassSourceManipulator.php:62]: nope... removed

**Comment by weaverryan** [comment]
Thanks again Jesse!

**Pull request merged**